pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java

```diff
@@ -1815,13 +1815,13 @@ public void testUnackedBlockAtBatch(int batchMessageDelayMs, boolean ackReceiptE
                     messages.add(msg);
                     totalReceiveMessages++;
                     consumer1.acknowledgeAsync(msg);
-                    log.info("Received message: " + new String(msg.getData()));
                 } else {
                     break;
                 }
             }
             // verify total-consumer messages = total-produce messages
-            assertEquals(totalProducedMsgs, totalReceiveMessages);
+            final int finalTotalReceiveMessages = totalReceiveMessages;
+            Awaitility.await().untilAsserted(() -> assertEquals(totalProducedMsgs, finalTotalReceiveMessages));
```

Comment:
Using awaitility seems useless here as it looks like you are comparing two constants

Reply:
Make a wrong modify. sorry for this. not this class